plugins/org.jkiss.dbeaver.erd.model/src/org/jkiss/dbeaver/erd/model/ERDPersistedState.java

```diff
@@ -96,10 +97,19 @@
             if (entitiesElem != null) {
                 // Parse data source
                 for (Element dsElem : XMLUtils.getChildElementList(entitiesElem, TAG_DATA_SOURCE)) {
+                    String dsProjectName = dsElem.getAttribute(ATTR_PROJECT);
+                    DBPProject project = projectMeta;
+                    if (!CommonUtils.isEmpty(dsProjectName)) {
+                        // Get owner project, if any
+                        project = DBWorkbench.getPlatform().getWorkspace().getProject(dsProjectName);
+                        if (project == null) {
+                            continue;
+                        }
+                    }
```

Comment:
I believe it makes sense to extract this code into a helper funtion.

Reply:
I don't think it deserves a helper function, since it is used just in one place and performs a particular job.

What the helper function will look like?

```java
public static DBPProject getProjectByName(@NotNull DBPProject defaultProject, @Nullable String projectName) {
    if (CommonUtils.isNotEmpty(projectName) {
        return DBWorkbench.getPlatform().getWorkspace().getProject(projectName);
    }
    return defaultProject;
}
```

I think it's pointless.